StreamChatClient_v3/WebSocketClient/WebSocketClient_Tests.swift

```diff
@@ -243,16 +244,42 @@ class WebSocketClient_Tests: StressTestCase {
         // Simulate connection
         test_connectionFlow()
         
-        let pingInterval = WebSocketClient.pingTimeInterval
+        // Simulate time longer than pingTimeInterval and assert `engine.sendPing` was called
         assert(engine.sendPing_calledCount == 0)
+        time.run(numberOfSeconds: WebSocketPingController.pingTimeInterval)
+        engine.simulatePong()
+        XCTAssertEqual(engine.sendPing_calledCount, 1)
+        
+        // Simulate time 3x longer than pingTimeInterval and assert 3 more pings
+        time.run(numberOfSeconds: WebSocketPingController.pingTimeInterval)
+        engine.simulatePong()
+        
+        time.run(numberOfSeconds: WebSocketPingController.pingTimeInterval)
+        engine.simulatePong()
+        
+        XCTAssertEqual(engine.sendPing_calledCount, 3)
+    }
+    
+    func test_pongTimeout() {
+        // Simulate connection
+        test_connectionFlow()
+        XCTAssertEqual(engine.connect_calledCount, 1)
+        
+        reconnectionStrategy.reconnectionDelay = 1
+        var timeInterval = WebSocketPingController.pingTimeInterval
         
         // Simulate time longer than pingTimeInterval and assert `engine.sendPing` was called
-        time.run(numberOfSeconds: pingInterval + 1)
+        assert(engine.sendPing_calledCount == 0)
+        time.run(numberOfSeconds: timeInterval)
         XCTAssertEqual(engine.sendPing_calledCount, 1)
+        timeInterval += WebSocketPingController.pongTimeoutTimeInterval
+        time.run(numberOfSeconds: timeInterval)
+        XCTAssertEqual(engine.disconnect_calledCount, 1)
+        XCTAssertFalse(webSocketClient.connectionState.isConnected)
         
-        // Simulate time 3x longer than pingTimeInterval and assert 3 more pings
-        time.run(numberOfSeconds: 3 * pingInterval)
-        XCTAssertEqual(engine.sendPing_calledCount, 1 + 3)
+        AssertAsync {
+            Assert.willBeTrue(self.engine.connect_calledCount == 2)
+        }
```

Comment:
These tests should be here at all. You just need to test the integration, not the functionality:
- You should test that `pingController.connectionStateDidChange` is called every time the connection state changes.
- Test that when the controller calls ping the engine is called with ping
- Test that when pong event is received, `pingController.pongReceived()` is called

Reply:
@VojtaStavik @b-onc 
I think this is the most important requested changes for this PR but it will required to rebuild some tests.
Should I do that? I think the current tests is more like component integration that cover all needed stuff.

Reply:
It's OK you need to rebuild the tests, because you moved the functionality out. The timing tests should be in the `PingController` tests, here should be just the tests for integration (as mentioned in my original comment).

Reply:
Fixed.